src/vinylcontrol/vinylcontrolxwax.cpp

```diff
@@ -793,22 +795,72 @@ bool VinylControlXwax::uiUpdateTime(double now) {
     return false;
 }
 
-void VinylControlXwax::establishQuality(bool quality_sample) {
-    m_bQualityRing[m_iQualPos] = quality_sample;
-    if (m_iQualFilled < QUALITY_RING_SIZE) {
-        m_iQualFilled++;
+int VinylControlXwax::getPositionQuality() {
+    int positionQualityPercent;
+
+    if (m_iPosition == -1) {
+        // No position code - happens when control vinyl is spinning slow (or during scratching) or the setup is incorrect
+        positionQualityPercent = 0;
+    } else if ((m_iPosition <= m_iQualityLastPosition - 5) ||
+            (m_iPosition >= m_iQualityLastPosition + 5)) {
+        // Position code changed not by more than 5. This indicates a normal spinning conrol vinyl.
+        positionQualityPercent = 100;
+    } else {
+        // Position code changed by more than 5. This indicates a fast spinning control vinyl, a jumping needle or a false signal.
```

Comment:
what if we use a simple calculation that gradually reduces the qualitypercent as the velocity increases?  Rather than a three-step threshold.

Reply:
I don't see, how this can be expressed by a linear expression:

- Position code -1 is a special case which needs to be handled
- +-5 codes difference happen in normal operation due to the sample rate of the quality information. If the vinyl spins a bit faster than 100% nominal speed, it's still a good signal, but we will have missing position codes.
- Other values, are either a extreme fast spinning vinyls, a jumping needle or a weak signal.